core/templates/services/playthrough-issues-backend-api.service.ts

```diff
@@ -15,87 +15,117 @@
 /**
  * @fileoverview Service for fetching issues and playthroughs from the backend.
  */
-require('domain/statistics/PlaythroughObjectFactory.ts');
-require('domain/statistics/PlaythroughIssueObjectFactory.ts');
-require('domain/utilities/url-interpolation.service.ts');
 
-require('services/services.constants.ajs.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient, HttpResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { throwError } from 'rxjs';
 
-angular.module('oppia').factory('PlaythroughIssuesBackendApiService', [
-  '$http', '$q', 'PlaythroughIssueObjectFactory', 'PlaythroughObjectFactory',
-  'UrlInterpolationService', 'FETCH_ISSUES_URL', 'FETCH_PLAYTHROUGH_URL',
-  'RESOLVE_ISSUE_URL',
-  function(
-      $http, $q, PlaythroughIssueObjectFactory, PlaythroughObjectFactory,
-      UrlInterpolationService, FETCH_ISSUES_URL, FETCH_PLAYTHROUGH_URL,
-      RESOLVE_ISSUE_URL) {
-    /** @type {PlaythroughIssue[]} */
-    var cachedIssues = null;
+import { PlaythroughIssueObjectFactory, PlaythroughIssue } from
+  'domain/statistics/PlaythroughIssueObjectFactory.ts';
+import { ServicesConstants } from 'services/services.constants.ts';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service.ts';
 
-    var getFullIssuesUrl = function(explorationId) {
-      return UrlInterpolationService.interpolateUrl(
-        FETCH_ISSUES_URL, {
-          exploration_id: explorationId
-        });
-    };
+@Injectable({
+  providedIn: 'root'
+})
+export class PlaythroughIssuesBackendApiService {
+  private cachedIssues = null;
 
-    var getFullPlaythroughUrl = function(expId, playthroughId) {
-      return UrlInterpolationService.interpolateUrl(
-        FETCH_PLAYTHROUGH_URL, {
-          exploration_id: expId,
-          playthrough_id: playthroughId
-        });
-    };
+  constructor(
+    private httpClient: HttpClient,
+    private playthroughIssueObjectFactory: PlaythroughIssueObjectFactory,
+    private urlInterpolationService: UrlInterpolationService) {}
 
-    var getFullResolveIssueUrl = function(explorationId) {
-      return UrlInterpolationService.interpolateUrl(
-        RESOLVE_ISSUE_URL, {
-          exploration_id: explorationId
+  // TODO(#7165): This has been marked any since marking explorationVersion
+  // to number throws an error. "Type 'number' is not assignable to type
+  // 'string | string[]'" and if this is marked to string it throws an
+  // error "TS2345: Argument of type '1' is not assignable to parameter
+  // of type 'string'".
+  fetchIssues(explorationId: string, explorationVersion: any): Promise<string> {
```

Comment:
Wasn't this `PlaythroughIssues[]` before? Seems you changed it to `string` by accident? Please change it back.

Reply:
I think it was like this from the beginning, still I have updated them. :)